src/util/storage/register_storage.hpp

```diff
@@ -84,10 +95,16 @@ namespace cyphal::support
 template <typename ResetPredicate>
 [[nodiscard]] std::optional<platform::storage::Error> save(platform::storage::interface::KeyValueStorage&            kv,
                                                            const registry::IIntrospectableRegistry& rgy,
+                                                           std::function<void()> const feed_watchdog_func,
```

Comment:
Since this is a template already, I don't think you would lose much by replacing std::function with a template argument type.